code/modules/projectiles/projectile.dm

```diff
@@ -691,6 +691,14 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 	proj.ammo.on_hit_mob(src, proj)
 	bullet_act(proj)
 
+/mob/living/carbon/human/do_projectile_hit(obj/projectile/proj)
+	. = ..()
+	new /obj/effect/temp_visual/dir_setting/bloodsplatter(loc, proj.dir)
+
+/mob/living/carbon/xenomorph/do_projectile_hit(obj/projectile/proj)
+	. = ..()
+	new /obj/effect/temp_visual/dir_setting/bloodsplatter/xenosplatter(loc, proj.dir)
+
```

Comment:
This will happen for all projectiles, including xeno spits and lasers
It might be better to add a `SPRAYS_BLOOD_ON_HIT` flag to the ammo and check it at the base of `/on_hit_mob()`, making things call parent

Reply:
AMMO_BALLISTIC covered the usecase without having to be overly specific.